rmate

```diff
@@ -381,11 +390,32 @@ function handle_connection {
 
 # connect to textmate and send command
 #
-exec 3<> "/dev/tcp/$host/$port"
+if [ -n "$unix" ] ; then
+    if [ ! -S "$unix" ] ; then
+        echo "$unix is not a socket!"
+        exit 1
+    fi
+
+    fifodir=$(mktemp -d) && \
+        mkfifo "$fifodir/in" "$fifodir/out" && \
+        ( (nc -U "$unix" < "$fifodir/in" > "$fifodir/out" &) &) && \
+        exec 4> "$fifodir/in" && \
+        exec 3< "$fifodir/out"
 
-if [ $? -gt 0 ]; then
-    echo "Unable to connect to TextMate on $host:$port"
-    exit 1
+    connect_status=$?
+
+    [ -d "$fifodir" ] && rm -rf "$fifodir"
```

Comment:
After everything is connected we can just trash the `fifodir` immediately, as everything has its file descriptors open and nothing else will need to refer to anything there by name again.